packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -800,6 +801,20 @@ export class Project extends ProjectBase {
               account: parsedArn?.account,
               region: parsedArn?.region,
             }));
+            // if secret comes from another account, SecretsManager will need to access
+            // KMS on the other account as well to be able to get the secret
+            if (parsedArn?.account !== stack.account) {
+              kmsIamResources.push(stack.formatArn({
+                service: 'kms',
+                resource: 'key',
+                resourceName: '*',
+                sep: '/',
+                // if we were given an ARN, we need to use the provided partition/account/region
+                partition: parsedArn?.partition,
+                account: parsedArn?.account,
+                region: parsedArn?.region,
```

Comment:
If you take my suggestion from above, this will become

```suggestion
                partition: parsedArn.partition,
                account: parsedArn.account,
                region: parsedArn.region,
```

Another good sign this is a needed change 🙂.